examples/simple-genai-server/main.go

```diff
@@ -103,7 +104,16 @@ func main() {
 		go tcpListener(*port, genAiConn)
 	} else {
 		log.Printf("Starting autonomous chat with Prompt: %s", *prompt)
-		go autonomousChat(*prompt, genAiConn, simConn, *numChats)
+		var wg sync.WaitGroup
+		wg.Add(1)
+		go autonomousChat(*prompt, genAiConn, simConn, *numChats, &wg)
+		wg.Wait()
+
+		log.Printf("Shutting down the Game Server.")
```

Comment:
I think it would be simpler if you just put these lines into the `autonomousChat` function when you hit the base case:

```
	if numChats <= 0 {
		log.Printf("Shutting down the Game Server.")
		shutdownErr := s.Shutdown()
		if shutdownErr != nil {
			log.Printf("Could not shutdown")
		}
		return
	}
```

I also think that having the wait group here means that the `<-sigCtx.Done()` line below never gets executed until after the autonomous chats complete. So an interrupt wouldn't be caught.

Reply:
I changed up the code to pass through the sigCtx to the autonomousChat.

We get an error `"error":"http: Server closed"` after the request to shutdown, but it looks like that's coming from the SDK Server httpEndpoint.

```
{"gsKey":"default/gen-ai-server-auto","message":"SDK server shutdown requested, waiting for game server shutdown","severity":"info","source":"*sdkserver.SDKServer","state":"RequestReady","time":"2024-02-23T23:25:18.158333903Z"}
{"message":"Shutting down SDK server","severity":"info","source":"main","time":"2024-02-23T23:25:18.196158733Z"}
{"error":"http: Server closed","message":"http server closed","severity":"info","source":"main","time":"2024-02-23T23:25:18.196415148Z"}
{"gsKey":"default/gen-ai-server-auto","message":"...shutting down workers","queue":"agones.dev.default.gen-ai-server-auto","severity":"info","source":"*sdkserver.SDKServer","time":"2024-02-23T23:25:18.196206956Z"}
{"error":"http: Server closed","gsKey":"default/gen-ai-server-auto","message":"Health check: http server closed","severity":"error","source":"*sdkserver.SDKServer","time":"2024-02-23T23:25:18.196492334Z"}
```